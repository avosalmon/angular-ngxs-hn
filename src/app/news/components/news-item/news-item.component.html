<div class="title">
  <a href="{{ news.url }}">{{ news.title }}</a>
</div>
<div class="info">
  <span>{{ news.points }} points </span>
  <span>by {{ news.user }} </span>
  <span>{{ news.time_ago }} </span>
  <span>| {{ news.comments_count }} comments</span>
</div>
