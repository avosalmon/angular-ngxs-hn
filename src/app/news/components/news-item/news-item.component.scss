.title {
  font-size: 1.1rem;
  padding-bottom: 8px;
}

.info {
  font-size: 0.75rem;
  color: #777;
  padding-bottom: 8px;
}
