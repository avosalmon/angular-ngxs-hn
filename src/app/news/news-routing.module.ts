import { NgModule } from '@angular/core';
import { Routes, RouterModule } from '@angular/router';
import { NewsComponent } from './news.component';
import { NewsListComponent } from './containers/news-list/news-list.component';

const routes: Routes = [
  {
    path: '',
    component: NewsComponent,
    children: [
      {
        path: '',
        component: NewsListComponent
      }
    ]
  }
];

@NgModule({
  imports: [RouterModule.forChild(routes)],
  exports: [RouterModule]
})
export class NewsRoutingModule { }
