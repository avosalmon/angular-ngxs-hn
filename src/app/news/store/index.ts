export * from './news.actions';
export * from './news.state';
