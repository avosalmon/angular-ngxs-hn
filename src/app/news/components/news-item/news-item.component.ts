import { Component, Input } from '@angular/core';
import { News } from '../../models/news.model';

@Component({
  selector: 'app-news-item',
  templateUrl: './news-item.component.html',
  styleUrls: ['./news-item.component.scss']
})
export class NewsItemComponent {

  @Input() news: News;

}
