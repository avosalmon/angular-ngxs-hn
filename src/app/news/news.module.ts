import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { HttpClientModule } from '@angular/common/http';
import { NgxsModule } from '@ngxs/store';

import { NewsRoutingModule } from './news-routing.module';
import * as store from './store';
import { NewsComponent } from './news.component';
import { NewsListComponent } from './containers/news-list/news-list.component';
import { NewsMaterialModule } from './news-material.module';
import { NewsItemComponent } from './components/news-item/news-item.component';
import { PaginationComponent } from './components/pagination/pagination.component';

@NgModule({
  declarations: [
    NewsComponent,
    NewsListComponent,
    NewsItemComponent,
    PaginationComponent
  ],
  imports: [
    CommonModule,
    HttpClientModule,
    NewsMaterialModule,
    NewsRoutingModule,
    NgxsModule.forFeature([store.NewsState])
  ]
})
export class NewsModule { }
