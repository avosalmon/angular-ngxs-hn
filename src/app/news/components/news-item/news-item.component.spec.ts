import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { News } from '../../models/news.model';
import { NewsItemComponent } from './news-item.component';

describe('NewsItemComponent', () => {
  let component: NewsItemComponent;
  let fixture: ComponentFixture<NewsItemComponent>;
  let news: News;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [NewsItemComponent]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(NewsItemComponent);
    component = fixture.componentInstance;

    news = {
      id: 1,
      title: 'The AI Threat to Open Societies',
      points: 42,
      user: 'malloryerik',
      time: 1548572441,
      time_ago: '3 hours ago',
      comments_count: 41,
      type: 'link',
      url: 'https://www.project-syndicate.org/onpoint/the-ai-threat-to-open-societies-by-george-soros-2019-01'
    };

    component.news = news;

    fixture.detectChanges();
  });

  test('create component', () => {
    expect(component).toBeTruthy();
  });
});
