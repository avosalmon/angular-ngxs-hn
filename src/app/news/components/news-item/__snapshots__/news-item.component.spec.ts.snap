// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsItemComponent should match snapshot 1`] = `
<app-news-item
  news={[Function Object]}
>
  <div
    class="title"
  >
    <a
      href="https://www.project-syndicate.org/onpoint/the-ai-threat-to-open-societies-by-george-soros-2019-01"
    >
      The AI Threat to Open Societies
    </a>
  </div><div
    class="info"
  >
    <span>
      42 points 
    </span>
    <span>
      by malloryerik 
    </span>
    <span>
      3 hours ago 
    </span>
    <span>
      | 41 comments
    </span>
  </div>
</app-news-item>
`;
